src/BWC/Component/JsBundle/Resources/public/js/filter.js

```diff
@@ -0,0 +1,114 @@
+if (typeof BWC == 'undefined') BWC = {};
+
+BWC.Filter = function(blacklist, whitelist) {
+    this.keys = BWC.Keys;
+    this.blacklist = [];
+    this.whitelist = [];
+
+    if (typeof blacklist != 'undefined' && blacklist) {
+        this.addToBlacklist(blacklist);
+    }
+
+    if (typeof whitelist !== 'undefined' && whitelist) {
+        this.addToWhitelist(whitelist);
+    }
+};
+
+BWC.Filter.prototype.isWhitelisted = function(key) {
```

Comment:
argument misnamed, it's not just a key, it can be anything, so ``value`` is more appropriate name... this method, and all others in ``BWC.Filter``